lib/Conversion/StructuredToMemref/StructuredToMemref.cpp

```diff
@@ -762,19 +761,43 @@ struct ScalarAddptrConverter : public OpConversionPattern<triton::AddPtrOp> {
         cast<triton::PointerType>(op.getPtr().getType()).getPointeeType();
     auto memrefType = MemRefType::get({1}, elemType, layout);
 
-    auto extractMetadataOp =
-        rewriter.create<memref::ExtractStridedMetadataOp>(loc, ptr);
-    auto base = extractMetadataOp.getBaseBuffer();
-    auto offset = extractMetadataOp.getOffset();
+    auto ptr = adaptor.getPtr();
+    auto definingOp = ptr.getDefiningOp();
```

Comment:
The code sequence you have here ends up asserting that the defining op is ReinterpretCastOp. Is that known to be true? is the else block the correct place to go if it is not, rather than an assert?

Reply:
Yeah this is known to be true. I have added a comment explaining why! Thanks for the comment as I realize this isn't clear and might confuse people.